### ISSUE TITLE
nRF53840-MDK (Dongle Version) Won’t Flash with new UF2 Bootloader

### ISSUE THREAD

**Issue by APDevice** [tinygo-org/tinygo#1147]
MakerDiary has replaced the Open Bootloader with the UF2 Bootloader. This appears to have broken existing support with the flash command.

`Open On-Chip Debugger 0.10.0
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
Info : auto-selecting first available session transport "swd". To override use 'transport select <transport>'.
cortex_m reset_config sysresetreq
adapter speed: 1000 kHz
Error: unable to find CMSIS-DAP device
Error: No Valid JTAG Interface Configured.
error: failed to flash /var/folders/lj/jy1t3gn147vfj8qy42z469tr0000gn/T/tinygo375609724/main.hex: exit status 255

**Comment by deadprogram** [comment]
Hi @APDevice do you happen to know the volume name of the board? I do not have one of these boards, but you should be able to get it working by changing this file https://github.com/tinygo-org/tinygo/blob/master/targets/nrf52840-mdk.json to make it more like a modified version of this https://github.com/tinygo-org/tinygo/blob/master/targets/circuitplay-bluefruit.json

We've love to have a PR that worked with the current bootloader!

**Comment by APDevice** [comment]
The volume name is “MDK-DONGLE”. 

I think I managed to set up the file right, though I apparently can’t test it as I’ve now I’ve run into an issue with installing from source.  (opened a separate issue for that)

**Comment by raqbit** [comment]
For me the volume comes up as `MDK-DONGLE` as well.

My `INFO_UF2.TXT`: 

```
UF2 Bootloader 0.2.13-42-g82464f9-dirty lib/nrfx (v2.0.0) lib/tinyusb (legacy-1500-g23df777b) s140 6.1.1
Model: nRF52840 MDK USB Dongle
Board-ID: nRF52840-Dongle-v1
Date: Feb  3 2020
```

**Comment by deadprogram** [comment]
Based on your description, the nrf52840-mdk.json should be changed to the following:

```json
{
    "inherits": ["nrf52840"],
    "build-tags": ["nrf52840_mdk"],
    "flash-1200-bps-reset": "true",
    "flash-method": "msd",
    "msd-volume-name": "MDK-DONGLE",
    "msd-firmware-name": "firmware.uf2",
    "uf2-family-id": "0xADA52840"
}
```

Can someone who has one of these boards try this out with the latest TinyGo `dev` branch?

**Comment by raqbit** [comment]
Now that I look at it, it does not look like the non-dongle [nrf52840-mdk](https://github.com/makerdiary/nrf52840-mdk) has the same bootloader as the [nrf52840-mdk-usb-dongle](https://github.com/makerdiary/nrf52840-mdk-usb-dongle). In fact, despite sharing a name these boards appear to be quite different, as the non-dongle version has DAPlink running on a separate MCU where the dongle-version used to be shipped with Nordic's Open Bootloader, which has been replaced with the UF2 bootloader for units produced after february of 2020.

Even though Makerdiary links to `machine/nrf52840-mdk` for tinygo support on the [dongle-variant wiki](https://wiki.makerdiary.com/nrf52840-mdk-usb-dongle/), I am not sure if this was actually tested and confirmed working.

**Comment by deadprogram** [comment]
I see. So actually adding a new board definition in TinyGo for the dongle is probably the correct move, since it also has a different pinout.

**Comment by deadprogram** [comment]
This has been released with v0.17.0 so now closing. Thanks!
